src/guidelines/contribute.md

```diff
@@ -0,0 +1,104 @@
+
+## Developing locally while consuming from CMS, Monorepo, Nora, etc.
+
+A common use case is you're working in CMS, monorepo, Nora, etc., and you discover a bug or missing feature
+in EDS. You'd like to be able to make a fix, but verify from the page or feature where you discovered the
+EDS issue. Here's a recommended process:
+
+- Cut a topic branch in EDS, make some changes and commit
+- Find the corresponding `package.json` file in the consumer app. For example, in monorepo it would be in [frontend/package.json](https://github.com/getethos/ethos/blob/cfe5122f12ce5ff84c00cda58fcc0b5d9f3d9534/frontend/package.json#L98)
+- Point to your new topic branch:
+
+```shell
+"ethos-design-system": "github:getethos/ethos-design-system#topic-branch"
+```
+
+- Now do `yarn install` to bring in the latest EDS pointing to your topic branch
+- Tricky part: if you are continuing to make commits to `topic-branch`, subsequent `yarn install`'s won't bring in your new commits! Here's the workaround:
+  - find the corresponding [yarn.lock](https://github.com/getethos/ethos/blob/cfe5122f12ce5ff84c00cda58fcc0b5d9f3d9534/yarn.lock) and look for something like:
+
+```shell
+"ethos-design-system@github:getethos/ethos-design-system#v1.1.42":
+version "1.1.42"
+resolved "https://codeload.github.com/getethos/ethos-design-system/tar.gz/a7ec68c3acb611f54d24b9abb737de3b40d848d4"
+dependencies:
+dayjs "^1.8.16"
+...
+```
+
+- Remove that whole stanza, save the lockfile, and rerun `yarn install`
+- Verify the sha1 commit pointed to in the updated lock file matches your last commit to EDS
+
+The above process is also documented in the [frontend/README.md](https://github.com/getethos/ethos/blob/cfe5122f12ce5ff84c00cda58fcc0b5d9f3d9534/frontend/README.md#ethos-design-system-aka-eds)
+
+You should now be able to restart your consumer application and reflect your latest EDS commits.
+
+Alternatively, you're absolutely welcome to try to leverage [yarn link](https://yarnpkg.com/lang/en/docs/cli/link/) which doesn't require an internet connection or the yarn.lock shenanigans described above. However, I had inconsistent results with linking local packages so I use the former approach.\_
+
+### Exposing Components
+
+In order for EDS components to be consumable via something like `import { MyComponent } from 'ethos-design-system'`, `MyComponent` will need to have been added to the component exports [index.js](https://github.com/getethos/ethos-design-system/blob/master/src/components/index.js) file.
+
+### Typescript Definitions
+
+If your consumer app is using Typescript, it will require that the component you wish to use has a corresponding `index.d.ts` Typescript [definition](https://github.com/getethos/ethos-design-system/blob/master/src/components/index.d.ts)—we will discuss this more in [How to submit your contribution](#how-to-submit-your-contribution) section that follows.
+
+### Form engine components
```

Comment:
@wafflehorse I did my best to at least provide some pointers to important concepts, files, and basically give an overview on all this. It turns out to be challenging to document without getting very long-winded. Would this section have helped you a couple weeks back and/or do you see an egregious omission?

Keep in mind these docs will hopefully evolve over time iteratively

Reply:
I think you nailed all the critical points. This would have definitely helped with the Select component. Thanks for adding on this!